RNTester/RNTester-macOS/TurboModuleCxx/winrt/Microsoft.ReactNative.MacOS.mm

```diff
@@ -0,0 +1,89 @@
+#include "Microsoft.ReactNative.h"
+#include <dispatch/dispatch.h>
+
+namespace winrt::Microsoft::ReactNative
+{
+
+struct MacOSUIDispatcher : implements<MacOSUIDispatcher, IReactDispatcher>
+{
+  bool HasThreadAccess() const noexcept
+  {
+    VerifyElseCrash(false);
+  }
+  void Post(const ReactDispatcherCallback& callback) noexcept
+  {
+    auto copy = callback;
+    dispatch_async(dispatch_get_main_queue(), ^{
+      copy();
+    });
```

Comment:
Btw, if helpful, I’m pretty sure that you can pass C++ lambda’s to libdispatch APIs.